File: migration/electron/shared/quickJumpTooltip.ts
import type { Translator } from './i18n'

export function getQuickJumpTooltip(
  key: string,
  enabled: boolean,
  targetName: string,
  basisName: string,
  t: Translator,
) {
  const group = key === '#'
    ? t('quickJump.symbolGroup')
    : t('quickJump.letterGroup', { key })

  return enabled
    ? t('quickJump.enabled', { target: targetName, basis: basisName, group })
    : t('quickJump.disabled', { target: targetName, basis: basisName, group })
}

File: lib/src/library/ui/quick_jump_tooltip.dart
import '../../i18n/app_i18n.dart';

String getQuickJumpTooltip({
  required String key,
  required bool enabled,
  required String targetName,
  required String basisName,
  required SmPlayerI18n i18n,
}) {
  final group =
      key == '#'
          ? i18n.t('quickJump.symbolGroup')
          : i18n.t('quickJump.letterGroup', {'key': key});

  return enabled
      ? i18n.t('quickJump.enabled', {
        'target': targetName,
        'basis': basisName,
        'group': group,
      })
      : i18n.t('quickJump.disabled', {
        'target': targetName,
        'basis': basisName,
        'group': group,
      });
}
